### PR TITLE
feat: add rule for redundant import alias

### PR DESCRIPTION
**Pull request by damif94**
PR for issue https://github.com/mgechev/revive/issues/853

A new rule `redundant-import-alias` that would warn upon redundant import aliases, i.e., when the imported package name matches the alias names.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -501,6 +501,7 @@ List of all available rules. The rules ported from `golint` are left unchanged a
 | [`use-any`](./RULES_DESCRIPTIONS.md#use-any)          |  n/a   |  Proposes to replace `interface{}` with its alias `any` |    no    |  no   |
 | [`datarace`](./RULES_DESCRIPTIONS.md#datarace)          |  n/a   |  Spots potential dataraces |    no    |  no   |
 | [`comment-spacings`](./RULES_DESCRIPTIONS.md#comment-spacings)          |  []string   |  Warns on malformed comments |    no    |  no   |
+| [`unused-import-alias`](./RULES_DESCRIPTIONS.md#unused-import-alias)          |  []string   |  Warns on import aliases matching the imported package name |    no    |  no   |
```
Comment (chavacava) [inline README.md:504]: the rule does no have arguments thus `[]string` should be replaced by n/a

File: rule/unused-import-alias.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	for _, imp := range file.AST.Imports {
+		if imp.Name == nil {
+			continue
+		}
+
+		if getImportPackageName(imp) == imp.Name.Name {
+			failures = append(failures, lint.Failure{
+				Confidence: 1,
+				Failure:    fmt.Sprintf("Import alias \"%s\" is not used", imp.Name.Name),
```
Comment (chavacava) [inline rule/unused-import-alias.go:26]:
```suggestion
				Failure:    fmt.Sprintf("Import alias \"%s\" is redundant", imp.Name.Name),
```

File: rule/unused-import-alias.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+			})
+		}
+	}
+
+	return failures
+}
+
+// Name returns the rule name.
+func (*UnusedImportAlias) Name() string {
+	return "unused-import-alias"
```
Comment (chavacava) [inline rule/unused-import-alias.go:38]: What about `redundant-import-alias` ?

**Review by chavacava — COMMENTED** [review]
Hi, thanks for the rule idea and its PR! 
I've left some comments, please let me know what do you think about them.

File: rule/unused-import-alias.go (hunk 3 above)
Comment (damif94) [reply rule/unused-import-alias.go:38]: Yes, i think its more accurate to call it "redundant"

**Comment by damif94** [comment]
> Hi, thanks for the rule idea and its PR! I've left some comments, please let me know what do you think about them.

I agree with your suggestions, "redundant" seems like more appropriate. I already did the changes.

What's your view on making this rule as default? It seems like a pretty unquestionable linting rule, but on the other hand I got fear of retrocompatibility on upgrading versions and not sure on what's the project policies on upgrades.

**Comment by chavacava** [comment]
> > Hi, thanks for the rule idea and its PR! I've left some comments, please let me know what do you think about them.
> 
> I agree with your suggestions, "redundant" seems like more appropriate. I already did the changes.
> 
> What's your view on making this rule as default? It seems like a pretty unquestionable linting rule, but on the other hand I got fear of retrocompatibility on upgrading versions and not sure on what's the project policies on upgrades.

Thanks for the modifications.
About making this rule default: historically, `revive` default rules were those of `golint`. I propose to let the new rule out of the defaults and actvite it through the config.

**Comment by chavacava** [comment]
Thanks @damif94 !

**Pull request merged**